### PR TITLE
update k8s components and injest reconciler runtime changes

### PR DESCRIPTION
**Pull request by katmutua**
### Key Updates
- Update k8s components to `v0.27.2`
- Update Reconciler Runtime to `v0.12.0`  https://github.com/vmware-labs/reconciler-runtime/releases/tag/v0.12.0
- Address `ioutil` deprecation and use similar functionality provided by package `io` or package `os` (https://pkg.go.dev/io/ioutil)

*Note*: No change in functionality is expected from this PR

### CODE REVIEW

File: pkg/binding/conventions_test.go
Comment (rashedkvm) [inline pkg/binding/conventions_test.go:1]: Line 23: `io/ioutil` has been deprecated, consider updating.

File: pkg/controllers/podintent_reconciler.go [hunk 1]
```diff
@@ -146,24 +144,29 @@ func ResolveConventions() reconcilers.SubReconciler {
...
-				key := tracker.NewKey(secretGVK, types.NamespacedName{Namespace: parent.Namespace, Name: ips.Name})
-				c.Tracker.Track(ctx, key, parentNamespacedName)
+				ref := tracker.Reference{
+					Kind:      secretGVK.Kind,
+					APIGroup:  secretGVK.Group,
+					Namespace: parent.Namespace,
+					Name:      ips.Name,
+				}
+				c.Tracker.TrackReference(ref, parent)
+				// c.Tracker.Track(ctx, key, parentNamespacedName)
```
Comment (rashedkvm) [inline pkg/controllers/podintent_reconciler.go:169]:
```suggestion

```

File: pkg/binding/conventions_test.go
Comment (katmutua) [reply pkg/binding/conventions_test.go:1]: @rashedkvm now using "os" and "io".

File: pkg/controllers/podintent_reconciler.go [hunk 2]
```diff
@@ -146,24 +144,28 @@ func ResolveConventions() reconcilers.SubReconciler {
...
 			conditionManager := parent.GetConditionSet().Manage(&parent.Status)
-			parentNamespacedName := types.NamespacedName{Namespace: parent.Namespace, Name: parent.Name}
 
 			var imagePullSecrets []string
 			for _, ips := range parent.Spec.ImagePullSecrets {
 				imagePullSecrets = append(imagePullSecrets, ips.Name)
 				// track ref for updates
-				key := tracker.NewKey(secretGVK, types.NamespacedName{Namespace: parent.Namespace, Name: ips.Name})
-				c.Tracker.Track(ctx, key, parentNamespacedName)
+				ref := tracker.Reference{
```
Comment (scothis) [inline pkg/controllers/podintent_reconciler.go:162]: much better, thanks

**Review by rashedkvm — APPROVED** [review]

**Review by tapint185151 — APPROVED** [review]

**Pull request merged**